### PR TITLE
[CITE-134] Add group to Authority Entry

### PR DESCRIPTION
**Pull request by radhikanyati94**
# Guidelines for Pull Requests

If you haven't yet read our code review guidelines, please do so, You can find them [here](https://diging.atlassian.net/wiki/spaces/DIGING/pages/2256076801/Code+Review+Guidelines).

Please confirm the following by adding an x for each item (turn `[ ]` into `[x]`).

- [x] I have removed all code style changes that are not necessary (e.g. changing blanks across the whole file that don’t need to be changed, adding empty lines in parts other than your own code)
- [x] I am not making any changes to files that don’t have any effect (e.g. imports added that don’t need to be added)
- [x] I do not have any sysout statements in my code or commented out code that isn’t needed anymore
- [x] I am not reformatting any files in the wrong format or without cause. 
- [x] I am not changing file encoding or line endings to something else than UTF-8, LF
- [x] My pull request does not show an insane amount of files being changed although my ticket only requires a few files being changed
- [x] I have added Javadoc/documentation where appropriate
- [x] I have added test cases where appropriate
- [x] I have explained any part of my code/implementation decisions that is not be self-explanatory

## Please provide a brief description of your ticket 
The group should be a citation group. Every user that has access to a group should be able to add/edit/delete authorities from that group. When a new authority is created a user should choose if that authority would be part of the citation group it is being created for (default) or just available to the user (no group).

An authority can belong to more than one group.

The "Manage Authority Entries" page should show the authorities categorized by group. E.g. a user would select a group and then see all authorities in that group.
https://diging.atlassian.net/browse/CITE-134
  
## Anything else the reviewer needs to know?

This PR can be merged only after the following PR is merged: https://github.com/diging/citesphere-model/pull/12

### CODE REVIEW

File: citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/AuthorityEntryController.java
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/AuthorityEntryController.java:155]: if `authorityGroups` is already initialized this line doesn't really do anything. In your if statement I would check `authorityEntry.get().getGroups() == null` and if so, initialize the list. Then just add the new group id. Also, make sure there are no duplicates in the list.

**Review by jdamerow — CHANGES_REQUESTED** [review]
- pom changes are missing
- and also, I released the model plugin, so you can use the release version now (it sometimes takes a while though till eclipse finds it in maven central).

File: citesphere/src/main/webapp/WEB-INF/views/auth/group/editItem.html
Comment (jdamerow) [inline citesphere/src/main/webapp/WEB-INF/views/auth/group/editItem.html:782]: since it's changing this line anyways, let's send the csrf token via the body.

**Review by jdamerow — CHANGES_REQUESTED** [review]

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Comment by jdamerow** [comment]
- the drop down on the manage authority page should use bootstrap classes
- let's put the checkbox to add an authority to the group above the results or it's not noticeable (especially if there are many results).
- let's add a column that shows the groups an authority belongs to on the authority list page

**Comment by jdamerow** [comment]
Make it so, Jenkins.

File: citesphere/src/main/webapp/WEB-INF/views/auth/group/editItem.html
Comment (jdamerow) [inline citesphere/src/main/webapp/WEB-INF/views/auth/group/editItem.html:1485]: Let's say "Add authority to group [group name]".

**Review by jdamerow — CHANGES_REQUESTED** [review]
on the authority list page, there needs to be an option to see all authorities that are not in a group.

**Comment by radhikanyati94** [comment]
Converted this pull request to draft because it can be merged only after https://github.com/diging/citesphere-model/pull/14 is merged. The build is failing because the model version mentioned in pom.xml doesn't exist.

File: citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AuthorityService.java
Comment (jdamerow) [inline citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AuthorityService.java:228]: the group manager has a get group method, doesn't it? why don't you use that one?

**Review by jdamerow — CHANGES_REQUESTED** [review]

File: citesphere/pom.xml
Comment (radhikanyati94) [reply citesphere/pom.xml:28]: Yes, but I have an open pull request in citesphere-model https://github.com/diging/citesphere-model/pull/14/files. Once it is merged, the version will be 1.8.

**Comment by jdamerow** [comment]
Ha, I thought I checked that but I guess I missed it 😳

---
Julia Damerow, PhD
Scientific Software Engineer
Global Biosocial Complexity Initiative
Digital Innovation Group
Arizona State University
email: ***@***.*** <>
web: http://diging.asu.edu, https://complexity.asu.edu/

**Comment by jdamerow** [comment]
Make it so, Jenkins.

File: citesphere/src/main/webapp/WEB-INF/views/auth/authorities/list.html
Comment (jdamerow) [inline citesphere/src/main/webapp/WEB-INF/views/auth/authorities/list.html:103]: This link doesn't work anymore now because it is relative and creates the wrong link when a group or the user has been selected.

Comment (jdamerow) [reply citesphere/src/main/webapp/WEB-INF/views/auth/authorities/list.html:103]: And neither does the delete link.

**Comment by jdamerow** [comment]
Make it so, Jenkins.

**Pull request merged**
